EvidenceApi/V1/Controllers/DocumentTypesController.cs

```diff
@@ -12,21 +13,49 @@ namespace EvidenceApi.V1.Controllers
     [ApiVersion("1.0")]
     public class DocumentTypesController : BaseController
     {
-        private readonly IDocumentTypeGateway _gateway;
-        public DocumentTypesController(IDocumentTypeGateway gateway)
+        private readonly IDocumentTypeGateway _documentTypeGateway;
+        private readonly IStaffSelectedDocumentTypeGateway _staffSelectedDocumentTypeGateway;
+
+        public DocumentTypesController(IDocumentTypeGateway documentTypeGateway, IStaffSelectedDocumentTypeGateway staffSelectedDocumentTypeGateway)
+        {
+            _documentTypeGateway = documentTypeGateway;
+            _staffSelectedDocumentTypeGateway = staffSelectedDocumentTypeGateway;
+        }
+
+        /// <summary>
+        /// Returns all recognised document types by team name
+        /// </summary>
+        /// <response code="200">OK</response>
+        /// <response code="404">Team cannot be found</response>
+        [HttpGet]
+        [Route("{team}")]
+        [ProducesResponseType(typeof(List<DocumentType>), StatusCodes.Status200OK)]
+        public IActionResult GetDocumentTypesByTeamName([FromRoute][Required] string team)
         {
-            _gateway = gateway;
+            var result = _documentTypeGateway.GetDocumentTypesByTeamName(team);
+            if (result.Count > 0)
+            {
+                return Ok(result);
+            }
+            return NotFound($"Cannot find team with name: {team}");
```

Comment:
What if it can find a team which has no document types? Maybe the message should be something like "No document types were found for team with name: {team}".

Reply:
Good point, addressed